weather.py

```diff
@@ -89,8 +202,137 @@ def generate_summary(weather_data):
+
+    # we extract the information with a loop
+
+    for information in weather_data:  # building list
+        date_list.append(information[0])
+        min_list.append(information[1])
+        max_list.append(information[2])
+
+
+    ## use the find_min to find the minimum temperature (AND ITS POSITION)
+    ## minimum date
+
+    min_and_position = find_min(min_list)  # our function gives us 2 values! position, and temperature
+
+    min_position = min_and_position[1]  # its a number e. g. 0
+    min_date = date_list[min_position]
+    min_date = convert_date(min_date) # convert the date into a human redeable format
+
+    # # min temperature
+
+    min_temp = min_and_position[0]  # extracting the minimum e.g. 49
+    min_temp = convert_f_to_c(min_temp)  # converting minimum to celsius e.g. 9.4
+    min_temp = format_temperature(min_temp)  # adding degree celsius
+
+
+    ## use the find_max to find the maximum temperature (AND ITS POSITION)
+
+    ## maximun date
+
+    max_and_position = find_max(max_list)  # our function gives us 2 values! position, and temperature
+
+    max_position = max_and_position[1]  # its a number e. g. 0
+    max_date = date_list[max_position]
+    max_date = convert_date(max_date) # convert the date into a human redeable format
+
+    # # max temperature
+
+    max_temp = max_and_position[0]  # extracting the minimum e.g. 49
+    max_temp = convert_f_to_c(max_temp)  # converting minimum to celsius e.g. 9.4
+    max_temp = format_temperature(max_temp)  # adding degree celsius
+
+    ## Average low temperature
+
+    min_temp_average = float(sum(min_list)/len(min_list))  #Give tha average low temperature in isotring format
+    min_temp_average= convert_f_to_c(min_temp_average) #we convert the average low temperature in celcius grades
+    min_temp_average = format_temperature(min_temp_average) #we format the isostring format into a redeable human format
+   
+
+
+
+    ## Average high temperature
+
+    max_temp_average = float(sum(max_list)/len(max_list)) #Give tha average max temperature in isotring format
+    max_temp_average= convert_f_to_c(max_temp_average) #we convert the average max temperature in celcius grades
+    max_temp_average = format_temperature(max_temp_average) # we format the isostring format into a redeable human format
+
+
+
+    #calculating the amount of days
+
+    days_amount = len(weather_data)
+
+    # CREATING THE OUTPUT SUMMARY
+
+    # summary = str(days_amount) + "Day Overview" + \n + 
+
+
+    # creating the first line of the summary
+
+    summary = str(days_amount) + " Day Overview"  # e.g. 5 Day Overview
+
+    # Adding the second line of code (min temperature value with the day that was min temperature)
+
+    summary += newline_twospaces  # one line down + 2 spaces
+    summary += "The lowest temperature will be "  # starting to write 2nd line
+    summary += min_temp
+    summary += ", and will occur on "
+    summary += min_date
+    summary += "."
```

Comment:
Nice use of string concatenation to build the summary report. Could also combine this with f-strings to condense into a single le, e.g.
```suggestion
    summary += f"The lowest temperature will be {min_temp} and will occur on {min_date}."
```